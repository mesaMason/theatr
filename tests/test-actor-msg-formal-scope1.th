dolphin(int weight, int length):
    receive:
        eat(int num):
            num = 5 + num
            print("eat()'s new num = ")
            print(num)
        swim(int num):
            num = 10 + num
            print("swim()'s new num = ")
            print(num)
    drop:
        print("inside drop()")
    after:
        int time

func main() -> int:
    actor d = new dolphin(5,10)
    print("Made dolphin")
    // Testing which scope of msg formal args with same name
    //hello(5) | d
    swim(55) | d
    // eat(1) | d
    // die() | d

    return 0
