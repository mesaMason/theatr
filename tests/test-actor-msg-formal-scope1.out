Made dolphin
inside drop()
swim()'s new num = 
65
eat()'s new num = 
6
